blocks/amp-story/block-selector.js

```diff
@@ -0,0 +1,92 @@
+
+const { __, sprintf } = wp.i18n;
+const { Component } = wp.element;
+const { Button } = wp.components;
+const {
+	dispatch,
+	select
+} = wp.data;
+const {
+	getBlock,
+	isBlockSelected,
+	hasSelectedInnerBlock,
+	getSelectedBlock
+} = select( 'core/editor' );
+const {
+	selectBlock
+} = dispatch( 'core/editor' );
+
+import LayerInserter from './layer-inserter';
+
+class BlockSelector extends Component {
+	render() {
+		if ( ! this.props.rootClientId ) {
+			return null;
+		}
+
+		const rootBlock = getBlock( this.props.rootClientId );
+
+		if ( ! rootBlock.innerBlocks.length ) {
+			return null;
+		}
+
+		let links = [];
+
+		window.lodash.forEachRight( rootBlock.innerBlocks, function( block, index ) {
+			let className = 'component-editor__selector';
+			if ( isBlockSelected( block.clientId ) || hasSelectedInnerBlock( block.clientId ) ) {
+				className += ' is-selected';
+			}
+
+			let title = sprintf( __( 'Layout %d ', 'amp' ), index + 1 );
+			if ( 'amp/amp-story-cta-layer' === block.name ) {
+				title = __( 'CTA Layer', 'amp' );
+			}
+			links.push(
+				<li className={ className } key={ 'selector-' + index }>
+					<Button onClick={ ( e ) => {
+						e.stopPropagation();
+						if ( getSelectedBlock.clientId !== block.clientId ) {
+							// @todo This selects the first inner child instead for some reason.
+							selectBlock( block.clientId );
+						}
+					}}>
+						{ title }
+					</Button>
+				</li>
+			);
+		} );
+
+		let className = 'component-editor__selector';
+		if ( isBlockSelected( this.props.rootClientId ) ) {
+			className += ' is-selected';
+		}
+
+		const inserterProps = {
+			rootClientId: this.props.rootClientId
+		};
+
+		links.push(
+			<li className={ className } key='page-selector'>
+				<Button onClick={ ( e ) => {
+					e.stopPropagation();
+					if ( getSelectedBlock.clientId !== this.props.rootClientId ) {
+						selectBlock( this.props.rootClientId );
+					}
+				}}>
+					{ __( 'Page', 'amp' ) }
+				</Button>
+			</li>
+		);
+
+		// @todo Creating a custom inserter since the default inserter doesn't allow taking the root client ID dynamically. Change if that becomes available.
+		return (
+			<ul className="editor-selectors">
+				<LayerInserter { ...inserterProps }/>
```

Comment:
Added a custom block inserter which just displays the relevant items in the list since it looks like the default `Inserter` is automatically taking the `clientId` of the selected block and then inserts a block after that. Since the list of layouts is fixed with the page block then this would mean that the inserter would always take the `amp-story-page` as the selected block and thus add the new block to the root level instead of inside the AMP Story page block (since the inserter inserts AFTER the selected block by default).

We could potentially create another issue in Gutenberg repo for allowing the `Inserter` to also accept the selected block ID as a param, however, it's likely not the most pressing issue right now and might take some time to get implemented.

(As of now) the custom inserter basically just displays the allowed blocks and that's it:
![screen shot 2018-08-22 at 9 37 58 pm](https://user-images.githubusercontent.com/3294597/44483628-459a3d80-a654-11e8-979a-33e38e479a46.png)

@westonruter Do you have any thoughts on if this makes sense for now?

Reply:
That sounds good!